test/model/content_test.dart

```diff
@@ -301,6 +301,21 @@ class ContentExample {
         '\n</code></pre></div>'),
     ]);
 
+  static const lineBreaksAfterCodeBlocks = ContentExample(
+    'blank text nodes after code blocks',
+    '    code block.\n\nsome content',
```

Comment:
Interesting! I wonder if this is the explanation for all the times we've seen this symptom. It hasn't felt super common in my experience — which would be consistent with that, because I think this four-space-indent syntax is not super commonly used in Zulip messages (because the triple-backticks syntax is usually more convenient).

Reply:
I attempted to replicate the issue by copying the messages mentioned in the report. I can confirm that this occurs exclusively when utilizing the four-space-indent syntax. As my attempts to reproduce it using triple-backticks were unsuccessful.

Additionally, it is noteworthy that I couldn't reproduce the issue when typing on the mobile app. It was only after sending the message from the web client that the double line breaks appeared.

Reply:
Upon further investigation, I've determined that this issue is associated with the 'trim' function utilized prior to dispatching the message in the mobile application:

```dart
// lib/widgets/compose_box.dart:69
@override
String _computeTextNormalized() {
  String trimmed = text.trim();
  return trimmed.isEmpty ? kNoTopicTopic : trimmed;
}
```

I'm uncertain whether this behavior should be altered to align with the web version or if it's acceptable to maintain its current state. I'd recommend creating a new issue to handle this. What are your thoughts, @chrisbobbe?

Reply:
Thanks for the investigation, @Khader-1! It sounds like you've found that the web client doesn't trim leading and trailing whitespace from the raw message content sent in the send-message request. Is that right?

(I think you mean the same method of `ComposeContentController`:

```dart
  @override
  String _computeTextNormalized() {
    return text.trim();
  }
```

The one you quoted belongs to `ComposeTopicController`.)

I've always assumed it's a good idea for the client to trim extra whitespace from the start and end of raw message content. It just seems neater that way, to me. If it's true that trimming is preferable (and I might be wrong), I would wonder why the web client doesn't do so. 🙂 I wonder if there's some Markdown syntax that requires leading or trailing whitespace and is defeated when the whitespace is removed.

If you want, you could raise the question of why the web app doesn't do trimming, and whether clients should trim or not, on [CZO](https://zulip.com/development-community/). Perhaps in the `#api design` stream, because it's about the expectations of the [send-message endpoint](https://zulip.com/api/send-message). But it seems like the answer doesn't need to block progress on the current issue.